kernel/gdt.c

```diff
@@ -41,15 +116,47 @@ static void gdt_set_gate(int num, uint32_t base, uint32_t limit, uint8_t access,
  */
 void gdt_init()
 {
-	gdtp.limit = (sizeof(t_gdt_entry) * 7) - 1;
+	gdtp.limit = (sizeof(t_gdt_entry) * GDT_SIZE);
 	gdtp.base = (uint32_t)gdt;
-	gdt_set_gate(0,	0,	0,	0,	0);	// NULL descriptor
-	gdt_set_gate(1,	0,	0xFFFF,	0x9A,	0xCF);	// Kernel Mode Code Segment
-	gdt_set_gate(2,	0,	0xFFFF,	0x92,	0xCF);	// Kernel Mode Data Segment
-	gdt_set_gate(3,	0,	0xFFFF,	0x96,	0xCF);	// Kernel Mode Stack Segment
-	gdt_set_gate(4,	0,	0xFFFF,	0xFA,	0xCF);	// User Mode Code Segment
-	gdt_set_gate(5,	0,	0xFFFF,	0xF2,	0xCF);	// User Mode Data Segment
-	gdt_set_gate(6,	0,	0xFFFF,	0x96,	0xCF);	// User Mode Stack Segment
+	gdt_set_gate(GDT_ENTRY_NULL,	GDT_ENTRY_NULL,		GDT_ENTRY_NULL,		GDT_ENTRY_NULL,							GDT_ENTRY_NULL);	// NULL descriptor
+	gdt_set_gate(GDT_KERNEL_CODE,	GDT_BASE_SEGMENT,	GDT_KERNEL_LIMIT,	(uint8_t)(GDT_KERNEL_CODE_ACCESS),		GDT_GRAN_BASE);		// Kernel Mode Code Segment
+	gdt_set_gate(GDT_KERNEL_DATA,	GDT_BASE_SEGMENT,	GDT_KERNEL_LIMIT,	(uint8_t)(GDT_KERNEL_DATA_ACCESS),		GDT_GRAN_BASE);		// Kernel Mode Data Segment
+	gdt_set_gate(GDT_KERNEL_STACK,	GDT_BASE_SEGMENT,	GDT_KERNEL_LIMIT,	(uint8_t)(GDT_KERNEL_STACK_ACCESS),		GDT_GRAN_BASE);		// Kernel Mode Stack Segment
+	gdt_set_gate(GDT_USER_CODE,		GDT_BASE_SEGMENT,	GDT_USER_LIMIT,		(uint8_t)(GDT_USER_CODE_ACCESS),			GDT_GRAN_BASE);		// User Mode Code Segment
+	gdt_set_gate(GDT_USER_DATA,		GDT_BASE_SEGMENT,	GDT_USER_LIMIT,		(uint8_t)(GDT_USER_DATA_ACCESS),			GDT_GRAN_BASE);		// User Mode Data Segment
+	gdt_set_gate(GDT_USER_STACK,	GDT_BASE_SEGMENT,	GDT_USER_LIMIT,		(uint8_t)(GDT_USER_STACK_ACCESS),			GDT_GRAN_BASE);		// User Mode Stack Segment
 	gdt_flush(&gdtp);
 }
 
+/* Print all the info from the gdt basic info and all of the entries
+ *
+ */
+void print_gdt()
+{
+	VGA_setforegroundcolor(VGA_COLOR_GREEN);
+	kprintf("GDT ADDRESS = 0x00000%x\n", gdt);
+	VGA_setforegroundcolor(VGA_COLOR_WHITE);
+	kprintf("GDT ENTRIES : \n");
+	kprintf("GDT Base = 0x%x\n", gdtp.base);
+	kprintf("GDT Limit = 0x%x\n", gdtp.limit);
+	uint8_t i = 1;
+	VGA_setforegroundcolor(VGA_COLOR_LIGHT_BROWN);
+	kprintf("BASE LOW | BASE MID | BASE HIGH | LIMIT LOW | GRAN | ACC\n");
+	VGA_setforegroundcolor(VGA_COLOR_WHITE);
+	kprintf("0x%x        ", gdt[0].base_low);
+	kprintf("0x%x        ", gdt[0].base_middle);
+	kprintf("0x%x        ", gdt[0].base_high);
+	kprintf("0x%x           ", gdt[0].limit_low);
+	kprintf("0x%x         ", gdt[0].granularity);
+	kprintf("0x%x\n", gdt[0].access);
+	while (i <= GDT_SIZE)
+	{
+		kprintf("0x%x        ", gdt[i].base_low);
+		kprintf("0x%x        ", gdt[i].base_middle);
+		kprintf("0x%x        ", gdt[i].base_high);
+		kprintf("0x%x        ", gdt[i].limit_low);
+		kprintf("0x%x        ", gdt[i].granularity);
+		kprintf("0x%x\n", gdt[i].access);
+		i++;
+	}
+}
```

Comment:
During my previous PR `kprintf` wasn't implemented to handle padding. Now that's the case and I think we should use it.